lib/Controller/SocialApiController.php

```diff
@@ -66,6 +72,38 @@ public function setAppConfig($key, $allow) {
 		return new JSONResponse([], Http::STATUS_OK);
 	}
 
+	/**
+	 * @NoAdminRequired
+	 *
+	 * update appconfig (user setting)
+	 *
+	 * @param {String} key the identifier to change
+	 * @param {String} allow the value to set
+	 *
+	 * @returns {JSONResponse} an empty JSONResponse with respective http status code
+	 */
+	public function setUserConfig($key, $allow) {
+		$userId = $this->userSession->getUser()->getUid();
```

Comment:
Might be better to get the user and check against it being null before getting the UID :)